check-log/lib/check-log.go

```diff
@@ -556,12 +555,13 @@ func saveState(f string, state *state) error {
 var errFileNotFoundByInode = fmt.Errorf("old file not found")
 
 func findFileByInode(inode uint, dir string) (string, error) {
-	fis, err := ioutil.ReadDir(dir)
+	entries, err := os.ReadDir(dir)
 	if err != nil {
 		return "", err
 	}
-	for _, fi := range fis {
-		if detectInode(fi) == inode {
+	for _, entry := range entries {
+		fi, err := entry.Info()
+		if err == nil && detectInode(fi) == inode {
 			return filepath.Join(dir, fi.Name()), nil
 		}
```

Comment:
Errors occurred in reading FileInfo will be ignored. Is it intended?

see fs.DirEntry

> If the file has been removed or renamed
> since the directory read, Info may return an error satisfying errors.Is(err, ErrNotExist).